contracts/wrapped_azero/lib.rs

```diff
@@ -148,15 +155,55 @@ pub mod wrapped_azero {
                     psp22: psp22::Data::default(),
                     metadata,
                     access_control,
+                    halted: false,
                 }
             } else {
                 panic!("Caller is not allowed to initialize this contract");
             }
         }
 
+        /// Emergency halt of all operations
+        ///
+        /// Can only be called by the contract's Admin.
+        #[ink(message)]
+        pub fn halt(&mut self) -> Result<()> {
+            let caller = self.env().caller();
+            let this = self.env().account_id();
+            self.check_role(caller, Role::Admin(this))?;
+            self.halted = true;
+
+            // emit event
+            Self::emit_event(self.env(), Event::Halted(Halted {}));
+
+            Ok(())
```

Comment:
it would be very cool to have a common interface/something for all haltable contracts (thank you captain obvious!)